pkg/hintrunner/zero/hint_reference_resolver.go

```diff
@@ -1,56 +1,45 @@
 package zero
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
 )
 
 type hintReferenceResolver struct {
-	refs []hintReference
-
-	numCellRefer    int
-	numResOperander int
+	refs map[string]hinter.Reference
 }
 
-type hintReference struct {
-	name string
-
-	cell      hinter.CellRefer
-	operander hinter.ResOperander
+func NewReferenceResolver() hintReferenceResolver {
+	refs := make(map[string]hinter.Reference)
+	return hintReferenceResolver{refs}
 }
 
-func (m *hintReferenceResolver) NumCellRefers() int { return m.numCellRefer }
-
-func (m *hintReferenceResolver) NumResOperanders() int { return m.numResOperander }
-
-func (m *hintReferenceResolver) AddCellRefer(name string, v hinter.CellRefer) {
-	m.refs = append(m.refs, hintReference{
-		name: name,
-		cell: v,
-	})
-	m.numCellRefer++
-}
-
-func (m *hintReferenceResolver) AddResOperander(name string, v hinter.ResOperander) {
-	m.refs = append(m.refs, hintReference{
-		name:      name,
-		operander: v,
-	})
-	m.numResOperander++
+func (m *hintReferenceResolver) AddReference(name string, v hinter.Reference) error {
+	shortName := shortSymbolName(name)
+	_, ok := m.refs[shortName]
+	if ok {
+		return fmt.Errorf("cannot overwrite reference %s (%s)", shortName, name)
+	}
+	fmt.Printf("hola1")
```

Comment:
Probably a debug leftover.

Reply:
I'm not sure being careful all the time is easier than removing those from time to time from the codebase. :D
I'm just trying to notice at least something useful while doing a review. :D